File: sdks/go/claims.go
package nucleus

import "github.com/golang-jwt/jwt/v5"

// NucleusClaims represents the JWT claims issued by Nucleus.
type NucleusClaims struct {
	jwt.RegisteredClaims

	Email          string                 `json:"email,omitempty"`
	FirstName      string                 `json:"first_name,omitempty"`
	LastName       string                 `json:"last_name,omitempty"`
	AvatarURL      string                 `json:"avatar_url,omitempty"`
	EmailVerified  bool                   `json:"email_verified,omitempty"`
	Metadata       map[string]interface{} `json:"metadata,omitempty"`
	OrgID          string                 `json:"org_id,omitempty"`
	OrgSlug        string                 `json:"org_slug,omitempty"`
	OrgRole        string                 `json:"org_role,omitempty"`
	OrgPermissions []string               `json:"org_permissions,omitempty"`
}

// UserID returns the subject claim, which is the Nucleus user ID.
func (c *NucleusClaims) UserID() string {
	return c.Subject
}

File: sdks/rust/src/claims.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

/// JWT claims issued by Nucleus.
///
/// These are extracted from a verified access token and contain user identity
/// information plus optional organisation context.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NucleusClaims {
    /// Subject — the Nucleus user ID.
    pub sub: String,
    /// Issuer.
    pub iss: String,
    /// Audience — the Nucleus project ID.
    pub aud: String,
    /// Expiration time (Unix timestamp).
    pub exp: u64,
    /// Issued-at time (Unix timestamp).
    pub iat: u64,
    /// Unique token identifier.
    #[serde(default)]
    pub jti: Option<String>,

    // ── User fields ──────────────────────────────────────────────
    #[serde(default)]
    pub email: Option<String>,
    #[serde(default)]
    pub first_name: Option<String>,
    #[serde(default)]
    pub last_name: Option<String>,
    #[serde(default)]
    pub avatar_url: Option<String>,
    #[serde(default)]
    pub email_verified: Option<bool>,
    #[serde(default)]
    pub metadata: Option<HashMap<String, serde_json::Value>>,

    // ── Organisation fields ──────────────────────────────────────
    #[serde(default)]
    pub org_id: Option<String>,
    #[serde(default)]
    pub org_slug: Option<String>,
    #[serde(default)]
    pub org_role: Option<String>,
    #[serde(default)]
    pub org_permissions: Option<Vec<String>>,
}

impl NucleusClaims {
    /// Returns the Nucleus user ID (the `sub` claim).
    pub fn user_id(&self) -> &str {
        &self.sub
    }
}
